builtin/provisioners/chef/linux_provisioner.go

```diff
@@ -10,10 +10,27 @@ import (
 )
 
 const (
-	chmod      = "find %s -maxdepth 1 -type f -exec /bin/chmod %d {} +"
+	chmodLinux = "find %s -maxdepth 1 -type f -exec /bin/chmod %d {} +"
+	chmodAix   = "find %[1]s ! -path %[1]s -prune -type f -exec /bin/chmod %[2]d {} +"
```

Comment:
Can you explain the difference between `find %[1]s ! -path %[1]s` and `find %s`? As I'm not familiar with the form. And is there a specific reason why you added the `-prune` option and omitted the `-maxdepth 1` option?

Reply:
In my research, the `%[N]s` notation allowed for positional parameters in formatting strings to be re-used. In this case, the values for `%[1]s` are the same (`linuxConfDir`), so instead of duplicating those values for each `%s` substitution. It basically lets me do this:

```
fmt.Sprintf(p.getChmodCommand(), linuxConfDir, 600)
```

instead of this (if I used `%s`)

```
fmt.Sprintf(p.getChmodCommand(), linuxConfDir, linuxConfDir, 600)
```

The other added benefit that I see is it allows the signature of `fmt.Sprintf` be the same for both AIX and Linux since both strings use the same values, just the AIX command re-uses one of those values more than the Linux command. If I went with the latter option of using `%s` then I would have to expose the logic of checking the os type when calling `Sprintf` to switch between the two and three parameter signature usage based on if I was running on AIX or Linux. 

As for `-maxdepth`, I removed it because the find command in AIX does not support it. So using this command as is was breaking when trying to provision the chef-client on our AIX systems. It looks to me the goal of the linux command is to find all files inside `/etc/chef` and make sure their permissions are set properly without diving into any subfolders. The use of `! -path %s -prune` accomplishes the same thing, just not as obvious. It basically prunes (removes) all results that are not in the `-path %s`.

Reply:
Thanks for the explanations! I already knew about `%[N]s`, but I wondered about the `! -path` and `-maxdepth` parts. But luckily you explained those as wel, so all good 👍